application/src/main/java/org/togetherjava/tjbot/config/RoleApplicationSystemConfig.java

```diff
@@ -0,0 +1,30 @@
+package org.togetherjava.tjbot.config;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+/**
+ * Represents the configuration for an application form, including roles and application channel
+ * pattern.
+ */
+public record RoleApplicationSystemConfig(
```

Comment:
Please could you document these fields. There's no mention anywhere else unfortunantly on what these do.